cmd/download_test.go

```diff
@@ -77,6 +77,65 @@ func TestDownloadWithoutFlags(t *testing.T) {
 	}
 }
 
+func TestSolutionFile(t *testing.T) {
+	testCases := []struct {
+		name, file, expectedPath, expectedURL string
+	}{
+		{
+			name:         "filename with special character",
+			file:         "special-char-filename#.txt",
+			expectedPath: "special-char-filename#.txt",
+			expectedURL:  "http://www.example.com/special-char-filename%23.txt",
+		},
+		{
+			name:         "filename with leading slash",
+			file:         "/with-leading-slash.txt",
+			expectedPath: fmt.Sprintf("%c%s", os.PathSeparator, "with-leading-slash.txt"),
```

Comment:
For readability, I'd like the hard coded parts to be part of the template:

```
fmt.Sprintf("%cwith-leading-slash.txt", os.PathSeparator)
```

Reply:
Done